alacritty/src/daemon.rs

```diff
@@ -55,6 +79,7 @@ where
             .stdin(Stdio::null())
             .stdout(Stdio::null())
             .stderr(Stdio::null())
+            .current_dir(foreground_process_path().unwrap_or_default())
```

Comment:
This is not a good idea, we should *conditionally* add the current dir, not fallback to an empty path that doesn't exist. Your `unwrap_or_default` here will actually prevent the subprocess from starting at all.

Reply:
Right, I thought that `PathBuf::default` pointed to a reasonable directory (which in hindsight does not make sense). I changed that and made `foreground_process_path` return a trait object error (like it is in master), because the Error might not be an `io::Error` for macos.